examples/neon-snake/src/logic/updatePlaying/checkLatestSectionForCollisions.ts

```diff
@@ -1,7 +1,46 @@
-import { Point, PlayerInfo, GameState } from "../types.ts"
-import { collisionGridPointer } from "../collisionGridHelpers.ts"
+import { Point, PlayerInfo, GameState, CollisionGrid, Snake } from "../types.ts"
+import {
+  collisionToGlobalPoint,
+  globalToCollisionPoint,
+} from "../collisionGridHelpers.ts"
 import { isLatestSectionOutOfBounds } from "../isLatestSectionOutOfBounds.ts"
 import { checkWinnersAndGameOver } from "../checkWinnersAndGameOver.ts"
+import { allowedCollisionPoints } from "../logicConfig.ts"
+
+function markCollisionGrid(
+  collisionGrid: CollisionGrid,
+  collisionPoint: Point,
+  snake: Snake,
+) {
+  //Initialize collision grid if it does not exist yet
+
+  if (!collisionGrid[collisionPoint.x]) {
+    collisionGrid[collisionPoint.x] = {}
+  }
+
+  collisionGrid[collisionPoint.x][collisionPoint.y] = true
+
+  if (!isCollisionPointRecentlyVisitedBySnake(snake, collisionPoint)) {
```

Comment:
I think you probably don't need to check for this again here

Reply:
This is a check to avoid pushing in same value multiple times. I'll add a comment

Reply:
Then can we reuse `isSameCollisionPointAsBefore`? I'm just worried about calling an array lookup (`.some()`) more than necessary.

Reply:
> Each markCollisionGrid call is done with different collisionPoint.

Then why do you need to "check to avoid pushing in same value multiple times."?

Reply:
Because the snake can be in the same cell multiple `update` calls in a row.